extensions/resource-deployment/src/interfaces.ts

```diff
@@ -41,6 +42,13 @@ export interface ResourceSubType {
 	provider: DeploymentProvider;
 	okButtonText?: OkButtonTextValue;
 	agreement?: AgreementInfo;
+	helpText?: HelpText;
+}
+
+export interface HelpText {
+	template: string;
+	links: azdata.LinkArea[];
```

Comment:
Links and when should be optional - they won't necessarily be needed for all cases